nixos/modules/services/desktops/playerctld.nix

```
@@ -0,0 +1,40 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.playerctld;
+in
+{
+
+  options = {
+
+    services.playerctld = {
+
+      enable = mkEnableOption (lib.mdDoc "Enable the playerctld daemon.");
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.playerctl ];
+
+    systemd.user.services.playerctld =
+      {
+        description = "Playerctld daemon to track media player activity";
+        wantedBy = [ "default.target" ];
+
+        serviceConfig = {
+          Type = "oneshot";
+          RemainAfterExit = true;
+          ExecStart = "${pkgs.playerctl}/bin/playerctld daemon";
+        };
+      };
+
+  };
```

```suggestion
    environment.systemPackages = [ pkgs.playerctl ];

    systemd.user.services.playerctld = {
      description = "Playerctld daemon to track media player activity";
      wantedBy = [ "multi-user.target" ];

      serviceConfig = {
        Type = "oneshot";
        RemainAfterExit = true;
        ExecStart = "${pkgs.playerctl}/bin/playerctld daemon";
      };
    };
  };
```

Can the playerctld stay in the foreground? Then we don't need oneshot and remainAfterExit which are a bit ugly.

---

yes it can. I modify this

---

done